test/utils/gcp-residency.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {strict as assert} from 'assert';
+import * as fs from 'fs';
+import * as os from 'os';
+
+import {SinonSandbox, SinonStub} from 'sinon';
+
+import * as gcpResidency from '../../src/gcp-residency';
+
+export class GCPResidencyUtil {
+  /**
+   * Stubs used in this utility. These are used within the provided sandbox.
+   * */
+  stubs: {
+    [key in
+      | 'fsReadFileSync'
+      | 'fsStatSync'
+      | 'processEnv'
+      | 'osNetworkInterfaces']?: SinonStub | void;
+  } = {};
+
+  constructor(public sandbox: SinonSandbox) {}
+
+  /**
+   * A simple utility for stubbing the networkInterface for GCE emulation.
+   *
+   * @param isGCE determines if the address should begin with `42:01` or not
+   */
+  setGCENetworkInterface(isGCE = true) {
+    const mac = isGCE ? '42:01:00:00:00:00' : '00:00:00:00:00:00';
+
+    this.stubs.osNetworkInterfaces ??= this.sandbox.stub(
+      os,
+      'networkInterfaces'
+    );
+    this.stubs.osNetworkInterfaces.returns({
+      'test-interface': [{mac} as os.NetworkInterfaceInfo],
+    });
+  }
+
+  /**
+   * A simple utility for stubbing the platform for GCE emulation.
+   *
+   * @param platform a Node.js platform
+   */
+  setGCEPlatform(platform: NodeJS.Platform = 'linux') {
+    this.sandbox.stub(os, 'platform').returns(platform);
+  }
+
+  /**
+   * A simple utility for stubbing the Linux BIOS files for GCE emulation.
+   *
+   * @param isGCE options:
+   *   - set `true` to simulate the files exist and are GCE
+   *   - set `false` for exist, but are not GCE
+   *   - set `null` for simulate ENOENT
+   */
+  setGCELinuxBios(isGCE: boolean | null) {
+    this.stubs.fsReadFileSync ??= this.sandbox.stub(fs, 'readFileSync');
+    this.stubs.fsStatSync ??= this.sandbox.stub(fs, 'statSync');
+
+    this.stubs.fsStatSync.callsFake(path => {
+      assert.equal(path, gcpResidency.GCE_LINUX_BIOS_PATHS.BIOS_DATE);
+
+      return undefined;
+    });
+
+    this.stubs.fsReadFileSync.callsFake((path, encoding) => {
+      assert.equal(path, gcpResidency.GCE_LINUX_BIOS_PATHS.BIOS_VENDOR);
+      assert.equal(encoding, 'utf8');
+
+      if (isGCE === true) {
+        return 'x Google x';
+      } else if (isGCE === false) {
+        return 'Sandwich Co.';
+      } else {
+        throw new Error("File doesn't exist");
+      }
+    });
+  }
+
+  removeServerlessEnvironmentVariables() {
```

Comment:
Not sure if it was intentional but this function and the one below are missing the JSDoc header.

Reply:
Good catch - I'll add